### PR TITLE
Fixes for animating between sibling scroll contexts

### PR DESCRIPTION
**Pull request by mattgperry**
For https://github.com/framer/company/issues/23054

### CODE REVIEW

File: src/projection/node/create-projection-node.ts [hunk 1]
```diff
@@ -748,16 +748,32 @@ export function createProjectionNode<I>({
...
+                        y: -node.scroll.y,
+                    })
+                }
+
                 if (!hasTransform(node.latestValues)) continue
                 transformBox(withTransforms, node.latestValues)
             }
 
-            if (hasTransform(this.latestValues)) {
+            if (!scrollOnly && hasTransform(this.latestValues)) {
```
Comment (shuangq) [inline src/projection/node/create-projection-node.ts:776]: I feel it makes the function a bit too complicated to keep track of what we should/should not do when scroll only, should we create a new helper for it?

File: dev/projection/shared-scroll-b-a.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            boxProjection.root.didUpdate()
+
+            setTimeout(() => {
+                matchViewportBox(box, boxOrigin)
+                matchViewportBox(newBox, boxOrigin)
+                matchViewportBox(box, newBox)
+            }, 50)
+        </script>
+    </body>
+</html>
```
Comment (shuangq) [inline dev/projection/shared-scroll-b-a.html:94]: should we also cover a test between two scroll screens?

File: src/projection/node/create-projection-node.ts [hunk 3]
```diff
@@ -748,11 +748,24 @@ export function createProjectionNode<I>({
             return boxWithoutScroll
         }
 
-        applyTransform(box: Box): Box {
+        applyTransform(box: Box, transformOnly = false): Box {
             const withTransforms = createBox()
             copyBoxInto(withTransforms, box)
-            for (let i = 0; i < this.path.length; i++) {
+            for (let i = 1; i < this.path.length; i++) {
```
Comment (shuangq) [inline src/projection/node/create-projection-node.ts:754]: Why do we skip the parent? Can we add a comment here to clarify?

Comment (mattgperry) [reply src/projection/node/create-projection-node.ts:754]: I think this was the root node but I've put it back to 0 as it doesn't really make a difference and is clearer

**Pull request merged**